src/query/plan/rewrite/index_lookup.hpp

```diff
@@ -522,6 +531,19 @@ class IndexLookupRewriter final : public HierarchicalLogicalOperatorVisitor {
 
   std::optional<LabelIx> FindBestLabelIndex(const std::unordered_set<LabelIx> &labels) {
     MG_ASSERT(!labels.empty(), "Trying to find the best label without any labels.");
+
+    for (const auto &[index_type, label, _] : index_hints_) {
+      if (index_type != IndexHint::IndexType::LABEL) continue;
+
+      if (!db_->LabelIndexExists(GetLabel(label))) {
+        throw QueryRuntimeException("Index for label {} doesn't exist", label.name);
+      }
+
+      if (labels.contains(label)) {
+        return label;
+      }
+    }
```

Comment:
you can preprocess the IndexHints once in the `PostProcessor` constructor and do all the maps you need

Reply:
Done.

NB: While the `IndexHints` struct that contains the maps is in preprocess.hpp (in order to be visible in index_lookup.hpp), 
 the preprocessing part is in planner.hpp since I get template-related linker errors otherwise.